### PR TITLE
Add a build command to felt

### PR DESCRIPTION
**Pull request by mdebbar**
The build command supports a `--watch` or `-w` flag to watch for changes and rebuild.

Fixes https://github.com/flutter/flutter/issues/40392

### CODE REVIEW

File: lib/web_ui/dev/build.dart [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          return pipeline.start();
+        }).then((_) {
+          if (runCount == _pipelineRunCount) {
+            print('*** Done! ***');
+          }
+        });
+        break;
+
+      case PipelineStatus.stopping:
+        // We are already trying to stop the pipeline. No need to do anything.
```
Comment (yjbanov) [inline lib/web_ui/dev/build.dart]: What will cause the pipeline to restart?

File: lib/web_ui/dev/build.dart [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        help: 'Run the build in watch mode so it rebuilds whenever a change'
+            'is made.',
+      );
+  }
+
+  @override
+  String get name => 'build';
+
+  @override
+  String get description => 'Build the web engine package.';
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/dev/build.dart]: I think we should agree on one name and stick to it. We are calling it "the Flutter web engine" elsewhere.

File: lib/web_ui/dev/build.dart [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import 'utils.dart';
+
+class BuildCommand extends Command<bool> {
+  BuildCommand() {
+    argParser
+      ..addFlag(
+        'watch',
+        abbr: 'w',
+        help: 'Run the build in watch mode so it rebuilds whenever a change'
+            'is made.',
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/dev/build.dart]: "file is saved" isn't technically correct. A file could be removed and that would also trigger a rebuild.

File: lib/web_ui/dev/build.dart [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+  void _onEvent(WatchEvent event) {
+    final String relativePath = path.relative(event.path, from: dir);
+    print('- [${event.type}] ${relativePath}');
+    if (_pipelineScheduled) {
+      return;
+    }
+    _pipelineRunCount++;
+    _pipelineScheduled = true;
+    Future.delayed(const Duration(milliseconds: 100)).then((_) {
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/dev/build.dart]: Great idea!

File: lib/web_ui/dev/build.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/dev/build.dart]: The previous switch case does that.

**Pull request merged**